lib/multi_reduce_quda.cu

```diff
@@ -31,6 +31,25 @@ namespace quda {
         return false;
       }
 
+      bool advanceBlockDim(TuneParam &param) const
+      {
+        if (TunableMultiReduction::advanceBlockDim(param)) {
+          return true;
+        } else {
+          if (param.block.z < n_batch && param.block.z < device::max_threads_per_block_dim(2) &&
+              param.block.x * param.block.y * (param.block.z + 1) <= device::max_threads_per_block() &&
+              param.block.z < max_n_batch_block_multi_reduce()) {
+            param.block.z++;
+            param.grid.z = (n_batch + param.block.z - 1) / param.block.z;
+            return true;
+          } else { // we have run off the end so let's reset
+            param.block.z = 1;
+            param.grid.z = (n_batch + param.block.z - 1) / param.block.z;
+            return false;
+          }
+        }
```

Comment:
Maybe this part can be abstracted in a function so it can be used else where?

Reply:
I'll see if I can move to the parent `TunableMultiReduction`.

Reply:
62a41b7